common/state/src/lib.rs

```diff
@@ -34,60 +34,73 @@ pub struct AgentState {
     offset_cf_opt: Options,
 }
 
+// This was the old new fn implementation. It could probably be broken up
+// into a few smaller, private functions.
+fn _construct_agent_state(path: &Path) -> Result<AgentState, StateError> {
```

Comment:
This was the original implementation of `AgentState::new` and was only being called by the main function. It seemed to make sense to move it into a module private function so we might be able to tackle breaking into some smaller bits at some point in the future. I was trying to avoid heaping more code into an already giant function.